netbox_branching/signal_receivers.py

```diff
@@ -25,9 +27,46 @@
     'handle_branch_event',
     'record_change_diff',
     'validate_branch_deletion',
+    'validate_branching_operations',
+    'validate_object_deletion_in_branch',
 )
 
 
+@receiver(post_clean)
+def validate_branching_operations(sender, instance, **kwargs):
+    """
+    Validate that branching operations are valid (e.g., not modifying deleted objects).
+    """
+    branch = active_branch.get()
+
+    # Only validate if we're in a branch and this model supports branching
+    if branch is None:
+        return
+
+    # Check if this model supports branching
+    object_type = ObjectType.objects.get(app_label=instance._meta.app_label, model=instance._meta.model_name)
+    if 'branching' not in object_type.features:
+        return
```

Comment:
It might be safer to use `ObjectType.objects.get_for_model()` here, since it wraps `.get_or_create()` and also contains the workaround for that signals/migration bug we saw in v4.4.0.